### ISSUE TITLE
ClassCastException: CNamed cannot be cast to CPrimitive

### ISSUE THREAD

**Issue by invd** [ultimate-pa/ultimate#378]
Observed in Ultimate 08d4a0152fab59f1da0c9942206f8aa51963d6ea after fixes in #359. Note that this only appears with the Bitvector setting, otherwise #377 is encountered.

Settings:
```
AutomizerMemDerefMemtrack.xml
svcomp-DerefFreeMemtrack-32bit-Automizer_Bitvector.epf
```

Error message:
```
 * Results from de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator:
  - ExceptionOrErrorResult: ClassCastException: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CNamed cannot be cast to de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CPrimitive
    de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator: ClassCastException: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CNamed cannot be cast to de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CPrimitive: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.expressiontranslation.BitvectorTranslation.convertIntToInt_NonBool(BitvectorTranslation.java:434)
RESULT: Ultimate could not prove your program: T
```

**Comment by Heizmann** [comment]
For which input program does the error occur?

**Comment by invd** [comment]
The affected target program is the one referred to  in #359, which is not public at the moment.

The newest Ultimate 1b2bd5cb23de2367ef85fe9d0051fe89b96beab6 still runs into the same problem:
```
 * Results from de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator:
  - ExceptionOrErrorResult: ClassCastException: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CNamed cannot be cast to de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CPrimitive
    de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator: ClassCastException: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CNamed cannot be cast to de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CPrimitive: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.expressiontranslation.BitvectorTranslation.convertIntToInt_NonBool(BitvectorTranslation.java:434)
```
Therefore the bugfix in 8a4d3c6c2a004a4c987b6c46337d1dba4dead967 was not (completely) successful.

**Comment by Heizmann** [comment]
But without a program I cannot reproduce the problem. 
What is our procedure for these cases? Could Daniel or anyone else extract a smaller program that we can make public?

**Comment by invd** [comment]
@danieldietsch and @jhoenicke have access to the program, but I agree that this situation is not optimal. In short, it was not anticipated that this program would be needed for reproducing regression issues this often since Ultimate has been able to run it a few months ago.

The program itself is a mangled, pre-processed version of the open source Trezor bootloader code from early 2018 that is hard to read, big (~179kB), microcontroller-specific , contains strings of local development paths via the line directives and lacks appropriate source and copyright notices.
We planned to provide a better version of the program, but things like Ultimate regressions and lack of time got in the way.

@Heizmann: I have sent you a partly cleaned up version of the mentioned program code via mail, more information can be found there. Thanks for your work on fixing these issues.

**Comment by danieldietsch** [comment]
I will add a minimized example. 

I also added a short tutorial for our delta debugger to our wiki: https://github.com/ultimate-pa/ultimate/wiki/Usage#deltadebugger-cli -- perhaps it allows others to already provide minimized examples.

**Comment by gschievelbein** [comment]
Fixed in commit 531c02fd63411bd00d58064d6226b9332f3aa410

**Comment by invd** [comment]
I can confirm that this error no longer appears in new Ultimate version 1516d6984f6ef049c75a980f3265d808ad37745d. Thanks!
